lib/core/src/persist/mod.rs

```diff
@@ -304,12 +322,57 @@ impl Persister {
     }
 }
 
+fn filter_to_where_clause(
+    type_filters: Option<Vec<PaymentType>>,
+    from_timestamp: Option<i64>,
+    to_timestamp: Option<i64>,
+) -> String {
+    let mut where_clause: Vec<String> = Vec::new();
+
+    if let Some(t) = from_timestamp {
+        where_clause.push(format!("ptx.timestamp >= {t}"));
```

Comment:
I think for pending receive swap we don't have a tx. How. would this filter behave on this case? Perhaps it should fallback to the swap.created_at ?

Reply:
I think at this point there will be no entry in the payment_tx_data table, so there's no payment to filter. Is that right @ok300?

Reply:
For pending receives, there will be a pseudo-tx created: https://github.com/breez/breez-sdk-liquid/blob/da43f7ff78498b3210015a2fa1075c14190b88c9/lib/core/src/receive_swap.rs#L262-L271

To cover this edge-case, it would be enough to set the `timestamp` to `Some(utils::now())` in that snippet (it's now `None`).

Reply:
Alternatively yes, the `receive_swap` table should have a `Created` entry, so the swap creation timestamp could be retrieved from there.

Reply:
I find it kind of messy because there are 3 swap tables joined in the query, so we have to check the created_at of each table. Setting the pseudo PaymentTxData timestamp to the ongoing_receive_swap.created_at (or now) has the same effect

Reply:
I think it makes sense to replace the None with now and then we don't need to fallback in the query.
Also setting to 'now' seems correct as this is the best time we know the transaction was broadcasted. 
What do you think @dangeross @ok300 ?

Reply:
Done [fbac4a2](https://github.com/breez/breez-sdk-liquid/pull/369/commits/fbac4a2c5f23ad70d4ae969df3ef67a09f74bba5)

Reply:
Looks good @dangeross but I think we should also change the `from_pending_swap` that puts the swap.created date on the payment because it is inconsistent with the new timestamp on the payment tx data.
Should we remove that method completely and use the `from_tx_data` instead now that we have the timestamp?

Reply:
I think the `from_pending_swap` is still needed to include swaps in the payments list that do not have an accompanying payment_tx_data entry (pseudo or otherwise).

Reply:
Aren't these the pending receives that you just added the timestamp for?

Reply:
Yeah right, looking at the SQL query and `sql_row_to_payment()` it will only ever return rows with payment_tx_data so this `match (tx, swap.clone())` is not needed
```
        match (tx, swap.clone()) {
            (None, None) => Err(maybe_tx_tx_id.err().unwrap()),
            (None, Some(swap)) => Ok(Payment::from_pending_swap(swap, payment_type)),
            (Some(tx), None) => Ok(Payment::from_tx_data(tx, None)),
            (Some(tx), Some(swap)) => Ok(Payment::from_tx_data(tx, Some(swap))),
        }
```
Does that sound right?

Reply:
That's what I thought. @ok300 can you confirm?

Reply:
I missed it the first time, there is a FULL JOIN between payment_tx_data and receive_swap, then LEFT JOINs on send_swaps and chain_swaps. So it will list receive_swaps without payment_tx_data and the match is valid.

I will add a coalesce to the timestamp filter to validate the first non null timestamp
```
coalesce(ptx.timestamp, rs.created_at) >= {t}
```

Reply:
I started with that and then got confused when we added the timestamp. Yes, I guess this is the right thing now.